bundle-workflow/tests/git/test_git_repository.py

```diff
@@ -0,0 +1,87 @@
+# Copyright OpenSearch Contributors.
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import subprocess
+import tempfile
+import unittest
+from unittest.mock import MagicMock
+
+from src.git.git_repository import GitRepository
+
+
+class TestGitRepository(unittest.TestCase):
+    def setUp(self):
+        self.repo = GitRepository(
```

Comment:
nit - Rather than cloning a dependency in this test, should we be mocking subprocess and ensuring it is invoked with the correct commands?

Reply:
You're right, it feels more like an integration test as is. But gets the job done, so leaving it as is for now in favor of covering more surface.